### ISSUE TITLE
Segmentation fail in chapter5 demo

### ISSUE THREAD

**Issue by Simpleyyt** [chukong/programmers-guide-samples#18]
I run the chapter 5 demo on Linux, and a segmentation fail error is occurred.

And I find the error in `AppDelegate.cpp`.

```
41     auto chapter5 = Chapter5();
42 
43     // run
44     director->runWithScene(chapter5.getScene1());
45 
46     return true;

```

`chapter5` is a local variable, and will be destroyed after `return`.

And you refer it in lambda of `Chapter5.cpp`:

```
 51     menuItem1->setCallback([&](cocos2d::Ref *sender) {
 52 
 53         Director::getInstance()->replaceScene(createScene2());
 54     });
```

**Comment by germanalen** [comment]
Wow, thank you.

**Comment by slackmoehrle** [comment]
This has now been fixed in the v3.8 branch.
